src/controller/CommissioningWindowOpener.cpp

```diff
@@ -175,23 +182,38 @@ CHIP_ERROR CommissioningWindowOpener::OpenCommissioningWindowInternal(Messaging:
 {
     ChipLogProgress(Controller, "OpenCommissioningWindow for device ID 0x" ChipLogFormatX64, ChipLogValueX64(mNodeId));
 
-    ClusterBase cluster(exchangeMgr, sessionHandle, mTargetEndpointId);
+    ClusterBase cluster(exchangeMgr, sessionHandle, mJointCommissioning ? 1 : mTargetEndpointId);
 
     if (mCommissioningWindowOption != CommissioningWindowOption::kOriginalSetupCode)
     {
         Spake2pVerifierSerialized serializedVerifier;
         MutableByteSpan serializedVerifierSpan(serializedVerifier);
         ReturnErrorOnFailure(mVerifier.Serialize(serializedVerifierSpan));
 
-        AdministratorCommissioning::Commands::OpenCommissioningWindow::Type request;
-        request.commissioningTimeout = mCommissioningWindowTimeout.count();
-        request.PAKEPasscodeVerifier = serializedVerifierSpan;
-        request.discriminator        = mDiscriminator.GetLongValue();
-        request.iterations           = mPBKDFIterations;
-        request.salt                 = mPBKDFSalt;
-
-        ReturnErrorOnFailure(cluster.InvokeCommand(request, this, OnOpenCommissioningWindowSuccess,
-                                                   OnOpenCommissioningWindowFailure, MakeOptional(kTimedInvokeTimeoutMs)));
+        if (mJointCommissioning == false)
```

Comment:
I was trying to correlate this code with the instructions section also:
`pairing open-commissioning-window 1 1 400 1000 1261 --joint-fabric 1`

So in the end the existing OCW command of the Administrator Commissioning Cluster is called but if the `--joint-fabric` is true then the call is redirected to OJCW of the Joint Fabric Administrator Cluster. imho, this is not the way the SDK is designed to work.

My expectation would be that the OJCW command of the Joint Fabric Administrator Cluster called directly:
`pairing open-joint-commissioning-window 1 1 400 1000 1261` So, in my view, it should be the other way around, where the OJCW ends up calling OCW. This will also allow to avoid adding JF code inside the existing Administrator Commissioning Cluster commands.

Reply:
Let' please test with the jfc and jfa applications versus the chip-tool or all-clusters-app.  These are the Joint Fabric test applications.

Reply:
Consolidated all JF-related logic into jf-control-app in the latest commit.

Reply:
As you can see, the existing flow (when mJointCommissioning is false) was never modified, and both the regular and the new flows were tested. The OpenJointCommissioningWindow command (when mJointCommissioning = true) functioned as expected, and the JF TXT appeared as anticipated.

I’m having some difficulty understanding your review comment. Are you suggesting that the current code (without the JF change) might contain an infinite loop? Are we potentially looking for an issue that isn’t there? If there is indeed an infinite loop, could you please provide more details or clarification? I’m happy to discuss it further.

Reply:
I made an error while reviewing the old flow. If this code was tested then I'm resolving the conversation.